gre & vti: use alternate way to check if kernel support is enabled

alinnastac opened the pull request:
When necessary support is built in kernel, gre and vti protocol
support is not enabled in netifd.

Signed-off-by: Alin Nastac <alin.nastac@gmail.com>

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

alinnastac committed:
gre: use alternative way to check if kernel support is enabled

When necessary support is built in kernel, gre protocol support is
not enabled in netifd.

Signed-off-by: Alin Nastac <alin.nastac@gmail.com>

obsy commented:
Use ||, not -o. See https://github.com/openwrt/openwrt/commit/1c0e13db43b41a1d527428b82bf88a032a916e84

alinnastac commented:
@obsy I fail to see the rationale behind shellcheck recommendation. AFAIK -o and -a are portable test operators (see man test).

Besides, using the syntax you propose will spawn 2 test processes instead of one.

obsy commented:
All scripts use this. So please use the same syntax as the rest.

alinnastac commented:
According to https://github.com/koalaman/shellcheck/wiki/SC2166, this recommendation was issued for generic test cases (e.g. test $1 -o $2), where left or right expressions might contain parenthesis.

In this particular case, none of the -o operands contain any parenthesis... But have it your way, I've pushed the variant you desired.

dedeckeh commented:
@alinnastac Please split up the gre and vti changes in separate commits and increase the PKG_RELEASE as well for each package

rsalvaterra commented on package/network/config/gre/files/gre.sh:
```diff
@@ -291,8 +291,6 @@ proto_grev6tap_init_config() {
-	[ -f /lib/modules/$(uname -r)/gre.ko ] && add_protocol gretap
-	[ -f /lib/modules/$(uname -r)/ip6_gre.ko ] && add_protocol grev6
-	[ -f /lib/modules/$(uname -r)/ip6_gre.ko ] && add_protocol grev6tap
+	[ -f /lib/modules/$(uname -r)/gre.ko ] || [ -d /sys/module/ip_gre ] && { add_protocol gre; add_protocol gretap; }
+	[ -f /lib/modules/$(uname -r)/ip6_gre.ko ] || [ -d /sys/module/ip6_gre ] && { add_protocol grev6; add_protocol grev6tap; }
```
I'd argue checking for the existence of the module files is completely wrong (think monolithic builds) and shouldn't even be done…

Commit:
vti: use alternative way to check if kernel support is enabled

When necessary support is built in kernel, vti protocol support is
not enabled in netifd.

Signed-off-by: Alin Nastac <alin.nastac@technicolor.com>

Commit:
xfrm: simplify the check for necessary kernel support

[ -d /sys/module/xfrm_interface ] is enough to check if
CONFIG_XFRM_INTERFACE support was enabled in kernel.

Signed-off-by: Alin Nastac <alin.nastac@technicolor.com>

dedeckeh commented:
Patches pushed to master; thx
